Controllers

Qmemorais opened the pull request.

Qmemorais committed:
ReloadProject

Reload project with some folders in DataL. Create Context to work with db. Create methods to work with classes db

Qmemorais committed:
RebuildBusinessL

set all dbset to on file

Qmemorais committed:
UnitOfWork

Add interfaces and UnitOfWork

Qmemorais committed:
D

fg

Qmemorais committed:
Reload Project UnitOfWork

rewrite UnitOfWork and interfaces

Qmemorais committed:
FinalServices

Qmemorais committed:
.

Qmemorais committed:
AddMigrationsToProject

Qmemorais committed:
Result_03.11

Qmemorais committed:
AddMapping

Qmemorais committed:
Merge pull request #5 from Qmemorais/Result_03.11

Result 03.11

Qmemorais committed:
AllMethodsUserController

Qmemorais committed:
.

anastasiia-ilkiv-eleks commented on Web_Music/Controllers/UserController.cs:
```diff
@@ -0,0 +1,104 @@
+        {
+            try
+            {
+                var users = _userService.GetAllUser();
```
_userService.GetAllUsers();

anastasiia-ilkiv-eleks replied:
before mapping users, check if the collection is not empty. If it is empty, return NotFound()

anastasiia-ilkiv-eleks commented on Web_Music/Controllers/UserController.cs:
```diff
@@ -0,0 +1,104 @@
+            try
+            {
+                var users = _userService.GetAllUser();
+                var getUser = _mapper.Map<IEnumerable<UserResponseModel>>(users);
```
better to name it mappedUsers

anastasiia-ilkiv-eleks commented on Web_Music/Controllers/UserController.cs:
```diff
@@ -0,0 +1,104 @@
+            }
+        }
+        [Route("Update")]
+        [HttpPut]
```
[HttpPut("{userId}")]

anastasiia-ilkiv-eleks commented on Web_Music/Controllers/UserController.cs:
```diff
@@ -0,0 +1,104 @@
+        }
+        [Route("Update")]
+        [HttpPut]
+        public IActionResult UpdateUser([FromBody] UserResponseModel requestModel)
```
Update method should accept Id separately from requestModel and use it in service as I mentioned in comments above. Also, here you use the response model in request. You should use UserUpdateRequestModel

public IActionResult UpdateUser([FromRoute] userId, [FromBody] UserUpdateRequestModel requestModel)

anastasiia-ilkiv-eleks commented on Web_Music/Models/PlaylistCreateRequestModel.cs:
```diff
@@ -0,0 +1,10 @@
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public int UserId { get; set; }
+        public UserCreateRequestModel UserCreateRequestModel { get; set; }
```
No need of UserCreateRequestModel, UserId is enough

Fix the same in other request models

anastasiia-ilkiv-eleks commented on Web_Music/Models/PlaylistCreateRequestModel.cs:
```diff
@@ -0,0 +1,10 @@
+{
+    public class PlaylistCreateRequestModel
+    {
+        public int Id { get; set; }
```
The same comments as in PlaylistCreateDto
Id shouldn't be here

anastasiia-ilkiv-eleks commented on Web_Music/Models/PlaylistUpdateRequestModel.cs:
```diff
@@ -0,0 +1,10 @@
+{
+    public class PlaylistUpdateRequestModel
+    {
+        public int Id { get; set; }
```
Apply all the comments for DTOs to RequestModels as well

anastasiia-ilkiv-eleks commented on Web_Music/SourceMappingProfile.cs:
```diff
@@ -0,0 +1,22 @@
+{
+    public class SourceMappingProfile : Profile
+    {
+        public SourceMappingProfile()
```
Separate profiles

anastasiia-ilkiv-eleks commented on Web_Music/Controllers/UserController.cs:
```diff
@@ -0,0 +1,104 @@
+        {
+            try
+            {
+                _userService.Delete(id);
```
before calling Delete method, check if this user exisits. If he doesn't exist, return NotFound()

anastasiia-ilkiv-eleks commented on Web_Music/Controllers/UserController.cs:
```diff
@@ -0,0 +1,104 @@
+        {
+            try
+            {
+                var user = _userService.GetUser(id);
```
before mapping user check 'user' variable for null. If it is null, return NotFound()

Qmemorais committed:
Update

Update from comments

Qmemorais committed:
FinishControllers

finish all methods in each controller

Qmemorais committed:
Refresh Models

Add new Tables Album,Artist. Update properties in other models. Create interfaces to UserPlaylist and PlaylistSong models

Qmemorais committed:
Previous Changes =)

Qmemorais committed:
Update BLL

Create new models Artist/Album. Create Services for them. Update other methods in services

Qmemorais committed:
Update Api Models

Add new Request/Response models to Artist/Albums. Add configuration this models to Startup. Add SourceMapping Artist/Album

Qmemorais committed:
Update Controllers

Update methods in userController/UserPlaylistController. Add new methods from Services like AddPlaylistTouser and GetAllUsersFromPlaylist

Qmemorais committed:
Update Controllers 2

Update methods in PlaylistController/PlaylistSongController. Add new method from Service like AddSongToPlaylist

Qmemorais committed:
Update PlaylistSongController

Add new method in IPlaylistService : getAllPlaylistsBySong(songId). Add this method to Controller

Qmemorais committed:
Update Song Controller

rename some values to better read. Update methods in Song Controller. Add method to get all songs in DB

Qmemorais committed:
Artist Controller

Create new Controller to work with Artist model and services

Qmemorais committed:
AlbumController

Create Album Controller to work with Album model and service

Qmemorais committed:
AlbumArtistController

Add Controller to get albums by one artist

Qmemorais committed:
SongService

Add methods to get list of songs by artist or album

Qmemorais committed:
Create ArtistSongController.cs

Add controler to get list of songs which exist in artist model

Qmemorais committed:
Create AlbumSongController.cs

create contoller to get list of songs from album

Qmemorais committed:
DB

Drop All Migrations. Create new

Qmemorais committed:
Update Services

Update models to have a List of others models. Update methods because it didn`t work

Qmemorais committed:
Change models

change models to add each model in another one. change methods

Qmemorais committed:
.

.

Qmemorais committed:
Change Services

Delete UnitOfWork and Generic Repository.  Update methods to work only with context

Qmemorais committed:
create unitTests

create Empty UnitTest Projects

Qmemorais committed:
UnitTest

Add UnitTests for all services and controllers. add tests to all UserServiceMethods and all UserControllerMethods

Qmemorais committed:
Update UserServiceTests.cs

update tests for userService. add new test addPlaylistToUser which already exist

Qmemorais committed:
Update UserControllerTests.cs

update verifying in tests of userController

Qmemorais committed:
UnitTests for UserPlaylistController

Create tests for UserPlaylistController

Qmemorais committed:
Update UserPlaylistControllerTests.cs

Update to call methods in tests

Qmemorais committed:
Update PlaylistControllerTests.cs

create rests for PlaylistController

Qmemorais committed:
UnitTests for PlaylistSongController

create tests for methods in PlaylistSongSongController

Qmemorais committed:
Update AlbumArtistController.cs

Update method to get albums from artist. add verifying does artist exist

Qmemorais committed:
Update AlbumArtistControllerTests.cs

create unit tests for methods of AlbumArtistController

Qmemorais committed:
Update ArtistSongController.cs

update methods to get all songs by artist. add verifying if artist exist

Qmemorais committed:
Update ArtistSongControllerTests.cs

create tests for method in ArtistSongController

Qmemorais committed:
Update AlbumSongController.cs

update method to get songs by album. add verifying if album exist

Qmemorais committed:
Update AlbumSongControllerTests.cs

create tests for method to get all songs by album in albumSongController

Qmemorais committed:
Update SongControllerTests.cs

create tests for methods in SongController

Qmemorais committed:
Update SongController.cs

update name of updating method

Qmemorais committed:
Update AlbumControllerTests.cs

create tests for methods in AlbumController

Qmemorais committed:
Update ArtistControllerTests.cs

create unit tests for methods ArtistController

Qmemorais committed:
Update PlaylistService.cs

update methods. change find method to get model from db to first.

Qmemorais committed:
Update PlaylistServiceTests.cs

create unit tests for each method in PlaylistService

Qmemorais committed:
Update SongService.cs

rewrite SongService methods to get song by Id

Qmemorais committed:
Update SongServiceTests.cs

create unit tests for SongService

Qmemorais committed:
Update AlbumService.cs

rewrite find method in db to first get model from db

Qmemorais committed:
Update AlbumServiceTests.cs

create tests for Album Service methods

Qmemorais committed:
Update ArtistService.cs

update find method to get data from db to first method from db

Qmemorais committed:
Update ArtistServiceTests.cs

create unit tests for ArtistServicce. create 2-3 test for each method

Qmemorais committed:
update Test Services

update names of DbSets in tests. add verifying some methods to assert the test

Qmemorais committed:
update Test Controllers

rename values to existId or existModel. add spaces between test methods
